extras/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisRecordProcessor.scala

```diff
@@ -135,18 +105,17 @@ private[kinesis] class KinesisRecordProcessor[T](
    */
   override def shutdown(checkpointer: IRecordProcessorCheckpointer, reason: ShutdownReason) {
     logInfo(s"Shutdown:  Shutting down workerId $workerId with reason $reason")
-    reason match {
+    // We want to return a checkpointer based on the shutdown cause. If we want to terminate,
```

Comment:
I dont get this. Dont we always want remove the checkpointer when shutdown() is called irrespective of the reason of shutdown?